src/components/SelectionList/CategorySelectorListItem.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {useState} from 'react';
+import useThemeStyles from '@hooks/useThemeStyles';
+import CategorySelector from '@pages/workspace/distanceRates/CategorySelector';
+import * as Policy from '@userActions/Policy/Policy';
+import BaseListItem from './BaseListItem';
+import type {InviteMemberListItemProps, ListItem} from './types';
+
+function CategorySelectorListItem<TItem extends ListItem>({item, onSelectRow, isFocused}: InviteMemberListItemProps<TItem>) {
+    const styles = useThemeStyles();
+    const [isCategoryPickerVisible, setIsCategoryPickerVisible] = useState(false);
+    const {policyID, groupID, categoryID} = item;
+
+    if (!policyID || !groupID) {
+        return;
+    }
+
+    const onSelect = (data: TItem) => {
+        setIsCategoryPickerVisible(true);
+        onSelectRow(data);
+    };
+
+    const setNewCategory = (selectedCategory: ListItem) => {
+        if (!selectedCategory.text) {
+            return;
+        }
+        Policy.setWorkspaceDefaultSpendCategory(policyID, groupID, selectedCategory.text);
```

Comment:
This component `CategorySelectorListItem.tsx` name seem a common reusable component, but the above logic not that. It is related to "Spend category". I think we need to move this logic out or rename the component to include "Spend" word. We can also change the file path